ReactNativeTemplate/android/app/src/main/AndroidManifest.xml

```diff
@@ -21,13 +21,47 @@
 			</intent-filter>
 		</activity>
 
-        <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />
+		<!-- IDP account picker activity -->
+		<!--
+            To enable this app as an IDP app that's capable of handling authentication
+            requests originating from other SP apps, uncomment the lines below and replace
+            'scheme' and 'host' with the values you wish to use to define your IDP app.
+            The URI constructed from this must be unique and will be set in the SP app
+            as well. This will be used to launch the IDP app from the SP app.
+
+            In the example given below, 'scheme' is "com.salesforce.reactnativetemplate" and
+            'host' is "authenticator". The full URI that needs to be set on the SP app would be
+            "com.salesforce.reactnativetemplate://authenticator".
+        -->
+		<!--
+        <activity android:name="com.salesforce.androidsdk.auth.idp.IDPAccountPickerActivity"
+            android:excludeFromRecents="true"
+            android:theme="@style/SalesforceSDK.ActionBarTheme">
+
+            <intent-filter>
+                <data android:scheme="com.salesforce.reactnativetemplate"
+                    android:host="authenticator" />
+                <action android:name="android.intent.action.VIEW" />
+                <category android:name="android.intent.category.BROWSABLE" />
+                <category android:name="android.intent.category.DEFAULT" />
+            </intent-filter>
+        </activity>
+        -->
 
+        <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />
 	</application>
 
     <uses-sdk android:minSdkVersion="21"
         android:targetSdkVersion="26" />
 
-    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
+    <!--
+        GCM permission to ensure that only this application can
+        receive the messages and registration result. This must be of
+        the format <app-package-name>.C2D_MESSAGE for it to work.
+    -->
+    <permission android:name="com.salesforce.reactnativetemplate.C2D_MESSAGE"
```

Comment:
Not sure why the `ReactNative` template was missing the GCM registration piece for push notifications, but I added it in.